icu4c/source/i18n/unicode/numberformatter.h

```diff
@@ -2392,6 +2397,8 @@ class U_I18N_API LocalizedNumberFormatter
  * The result of a number formatting operation. This class allows the result to be exported in several data types,
  * including a UnicodeString and a FieldPositionIterator.
  *
+ * Instances of this class are immutable and thread-safe.
```

Comment:
Except for move assignment. Not worth mentioning, though. It would just be confusing.